### PR TITLE
Add filters to catalog

### PR DESCRIPTION
**Pull request by ThePletch**

### CODE REVIEW

File: app/assets/stylesheets/gift_requests.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  }
+}
+.filter-block {
+  display: inline-block;
+  margin: 10px 20px;
+  input {
+    margin: 10px;
+  }
+
+  input[type="range"] {
```
Comment (ThePletch) [inline app/assets/stylesheets/gift_requests.scss]: got rid of the range, gonna nix this

File: app/controllers/gift_requests_controller.rb [hunk 2]
```diff
@@ -45,7 +45,18 @@ def catalog
     @back_route = events_path
 
     @event = Event.find(params[:id])
-    @gift_requests = @event.gift_requests.unreserved.order(created_at: :asc).page(params[:page] || 1)
+    @gift_requests = @event.gift_requests.unreserved.order(created_at: :asc)
+    if params[:gender] && ["M", "F"].include?(params[:gender])
```
Comment (ThePletch) [inline app/controllers/gift_requests_controller.rb]: `include?` is nil-safe, so i can get rid of the first part of this conditional

File: app/controllers/gift_requests_controller.rb [hunk 3]
```diff
@@ -45,7 +45,18 @@ def catalog
...
+      @gift_requests = @gift_requests.where(gender: params[:gender])
+    end
+
+    if params.slice(:min_age, :max_age).any?
+      @gift_requests = @gift_requests.where("age BETWEEN ? AND ?",
+        params[:min_age].present? ? params[:min_age] : 0,
```
Comment (ThePletch) [inline app/controllers/gift_requests_controller.rb]: would be better done as `params[:min_age].presence || 0`, ditto to line below it

File: app/assets/stylesheets/gift_requests.scss [hunk 4]
```diff
@@ -0,0 +1,27 @@
+.filter-box {
+  background-color: #eee;
+  margin: 5px;
+  border: 1px solid #ccc;
+  border-radius: 5px;
+  padding: 1em;
+
+  h4, h5 {
+    font-weight: bold;
```
Comment (jonathanchiu) [inline app/assets/stylesheets/gift_requests.scss]: Minor CSS nit, but Bootstrap provides its own `<strong>` CSS rule that makes the font-weight bold already so you can use that instead, and nest it in the header tags, and just remove this CSS rule.

File: app/views/gift_requests/catalog.html.slim [hunk 5]
```diff
@@ -1,5 +1,23 @@
...
+    .filter-box
+      h4 Filters
+      = form_tag catalog_event_path(id: @event.id), method: :get do
+        .filter-block.gender-filters
+          h5 Gender
+          = radio_button_tag :gender, "M", params[:gender] == "M"
+          = label_tag :gender_male, "Male"
+          = radio_button_tag :gender, "F", params[:gender] == "F"
+          = label_tag :gender_female, "Female"
+          = radio_button_tag :gender, "any", !["M", "F"].include?(params[:gender])
```
Comment (jonathanchiu) [inline app/views/gift_requests/catalog.html.slim]: `exclude?` instead of `include?` perhaps to avoid the bang?

**Comment by jonathanchiu** [comment]
Just 2 incredibly minor comments, looks solid overall

File: app/views/gift_requests/catalog.html.slim (hunk 5 above)
Comment (ThePletch) [inline app/views/gift_requests/catalog.html.slim]: forgot `exclude?` existed, good call

**Comment by caramcc** [comment]
Aside from what you pulled out, everything looks 👌

**Pull request merged**